### ISSUE TITLE
Collapsing comments within expressions

### ISSUE THREAD

**Issue by stevegutz** [jhipster/prettier-java#352]
**Prettier-Java 0.5.1**

The following input causes prettier to produce output that is both in correct and that changes when prettier is run again. It seems as though I need to run `prettier --write` 4 times before it reaches the state shown in the output. 

**Input:**

```java
public class InlineComment {

  public boolean method() {
    return "some string value".equals("some string value")
        // This is a very long multiline comment that appears in the middle of an expression:
        // this seems to cause problems
        // for prettier.
        || "some other string value".equals("some other string value");
  }
}
```

**Output:**

```java
public class InlineComment {

  public boolean method() {
    return (
      "some string value".equals("some string value") || // for prettier. // this seems to cause problems // This is a very long multiline comment that appears in the middle of an expression:
      "some other string value".equals("some other string value")
    );
  }
}
```

**Expected behavior:**

The javascript equivalent maintains the long comments:
```js
function method() {
  return (
    "some string value" === "some string value" ||
    // This is a very long multiline comment that appears in the middle of an expression:
    // this seems to cause problems
    // for prettier.
    "some other string value" === "some other string value"
  );
}
```

The seems very reasonable to me. I'd say that whatever happens should require a single execution of prettier.

If comments are still collapse, they should be in the correct order. In such a case, I think it would also be nice to trim the leading `// ` when doing so, but I understand that that may be difficult.

**Comment by clementdessoude** [comment]
Hi @stevegutz,

Thank you for reporting this! Just to let you know, I started to work on a fix in #353, but there is still some work to be done.

**Comment by stevegutz** [comment]
Thanks!
